### ISSUE TITLE
GitRepository commit processing is brittle

### ISSUE THREAD

**Issue by davisjam** [mauricioaniche/repodriller#120]
`GitRepository` has these limits:

```java
private static final int MAX_SIZE_OF_A_DIFF = 100000;
private static final int DEFAULT_MAX_NUMBER_OF_FILES_IN_A_COMMIT = 5000;
```

If you set these values both to 1, you will get RuntimeExceptions while building and/or processing ChangeSets.

See also #118 and #119 where I noticed this bug as reported by @AjroudRami.

**Comment by mauricioaniche** [comment]
Yes, they were introduced due to crazy large commits that may exist in repos and could throw OOM errors. 

The user can change the configuration if she wants to, though. What we can try to do is to be friendly and warn her if limits are too low.
